src/Microsoft.AspNet.Identity/IdentityBuilder.cs

```diff
@@ -159,7 +159,9 @@ public virtual IdentityBuilder AddDefaultTokenProviders()
         /// <returns>The <see cref="IdentityBuilder"/>.</returns>
         public virtual IdentityBuilder AddUserManager<TUserManager>() where TUserManager : class
```

Comment:
This makes me wonder what happens if the user passes UserManager<UserType> as the argument (I imagine it could cause a cycle in DI) and also in general some TUserManager that does not derive from UserManager<UserType> (I imagine it would fail later on but the exception could be not that great :smile:). 

I remember we had to not have generic constraints in this method because we had to make IdentityBuilder non-generic but I am wondering if we should add more runtime type checks to throw nicer exceptions.